annotator/src/main/java/uk/ac/ebi/gxa/annotator/loader/annotationsrc/BioMartAnnotationSourceLoader.java

```
@@ -0,0 +1,306 @@
+            annotationSource.setMySqlDbUrl(getProperty(MYSQLDBURL_PROPNAME, properties));
+
+            updateBioMartProperties(properties, annotationSource);
+
+            updateBioMartArrayDesigns(properties, annotationSource);
+
+        } catch (IOException e) {
+            throw new AnnotationLoaderException("Cannot read annotation properties", e);
+        }
+
+        return annotationSource;
+    }
+
+    protected void writeSource(BioMartAnnotationSource annSrc, Writer out) {
+        PropertiesConfiguration properties = new PropertiesConfiguration();
+        properties.addProperty(ORGANISM_PROPNAME, annSrc.getOrganism().getName());
+        properties.addProperty(SOFTWARE_NAME_PROPNAME, annSrc.getSoftware().getName());
+        properties.addProperty(SOFTWARE_VERSION_PROPNAME, annSrc.getSoftware().getVersion());
+        properties.addProperty(URL_PROPNAME, annSrc.getUrl());
+        properties.addProperty(DATABASE_NAME_PROPNAME, annSrc.getDatabaseName());
+        properties.addProperty(DATASET_NAME_PROPNAME, annSrc.getDatasetName());
+        properties.addProperty(MYSQLDBNAME_PROPNAME, annSrc.getMySqlDbName());
+        properties.addProperty(MYSQLDBURL_PROPNAME, annSrc.getMySqlDbUrl());
+
+        //Write bioentity types
+        StringBuffer types = new StringBuffer();
+        int count = 1;
+        for (BioEntityType type : annSrc.getTypes()) {
+            types.append(type.getName());
+            if (count++ < annSrc.getTypes().size()) {
+                types.append(",");
+            }
+        }
+        properties.addProperty(TYPES_PROPNAME, types.toString());
+
+        writeBioMartProperties(annSrc, properties);
+
+        writeBioMartArrayDesign(annSrc, properties);
+
+        try {
+            properties.save(out);
+
+        } catch (ConfigurationException e) {
+            log.error("Cannot write Annotation Source " + annSrc.getAnnotationSrcId(), e);
```

Could you propagate this exception to the admin GUI via ErrorResult rather than just logging it? It would save the user having to trawl through the log files to see what happened.